widget/playground/src/components/SingleList/SingleList.tsx

```diff
@@ -0,0 +1,176 @@
+import type { PropTypes } from './SingleList.types';
+import type { ForwardedRef } from 'react';
+import type { CommonProps } from 'react-window';
+
+import {
+  CloseIcon,
+  Divider,
+  IconButton,
+  Image,
+  ListItemButton,
+  Radio,
+  RadioRoot,
+  SearchIcon,
+  TextField,
+  Typography,
+  VirtualizedList,
+} from '@rango-dev/ui';
+import React, { forwardRef, useEffect, useState } from 'react';
+
+import {
+  HeaderContainer,
+  IconWrapper,
+  RadioList,
+  StyledButton,
+} from './SingleList.styles';
+
+const PAGE_SIZE = 20;
+const ITEM_HEIGHT = 46;
+const PADDING_SPACE = 8;
+
+export function SingleList(props: PropTypes) {
+  const { list, onChange, defaultValue, title, icon, searchPlaceholder } =
+    props;
+  const [virtualList, setVirtualList] = useState(list);
+  const [hasNextPage, setHasNextPage] = useState(true);
+
+  const [searchValue, setSearchValue] = useState('');
+  const [item, setItem] = useState('');
+  const filteredList = searchValue
+    ? virtualList.filter((item) =>
+        item.name.toLowerCase().includes(searchValue.toLowerCase())
+      )
+    : list;
+
+  const handleConfirm = () => {
+    if (item) {
+      onChange(item);
+    }
+  };
+
+  useEffect(() => {
+    setHasNextPage(filteredList.length > virtualList.length);
+  }, [virtualList.length]);
+  useEffect(() => {
+    setVirtualList(filteredList.slice(0, PAGE_SIZE));
```

Comment:
Searching in the list of tokens is not correct, the entire list should be searched

Reply:
fixed.